packages/functional-tests/tests/subscription-tests/subscription.spec.ts

```diff
@@ -228,9 +229,12 @@ test.describe('severity-2 #smoke', () => {
 
       await relier.goto();
       const subscribeUrl = await relier.getUrl();
+      expect(subscribeUrl).toBeDefined();
       expect(subscribeUrl, 'Subscribe button has no href.').toBeTruthy();
 
-      const rpSearchParamsBefore = relier.getRpSearchParams(subscribeUrl);
+      const rpSearchParamsBefore = relier.getRpSearchParams(
+        subscribeUrl as string
```

Comment:
I see why you had to do some of the type casting this way, I know we can't throw here or something because of no conditionals in tests. I wish it was smart enough to interpret this correctly but it is what it is I suppose.

Reply:
At least it calls attention to the fact that we should probably add an assertion to make sure things are in fact defined. I'm a bit surprised that the assertion doesn't work as type guard, I agree that would be nice. Maybe someday.